e2e: retry creation of the RBD StorageClass

nixpanic opened the pull request:
On occasion the creation of the StorageClass can fail due to an
etcdserver timeout. If that happens, the creation can be attempted after
a delay.

This has already been done for CephFS StorageClasses, but was missed for
RBD.

See-also: ceph/ceph-csi@8a0377ef02

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

nixpanic committed:
e2e: retry creation of the RBD StorageClass

On occasion the creation of the StorageClass can fail due to an
etcdserver timeout. If that happens, the creation can be attempted after
a delay.

This has already been done for CephFS StorageClasses, but was missed for
RBD.

See-also: ceph/ceph-csi@8a0377ef02
Signed-off-by: Niels de Vos <ndevos@redhat.com>

Madhu-1 commented on e2e/rbd_helper.go:
```diff
@@ -167,9 +168,25 @@ func createRBDStorageClass(
+	return wait.PollImmediate(poll, timeout, func() (bool, error) {
+		_, err = c.StorageV1().StorageClasses().Create(context.TODO(), &sc, metav1.CreateOptions{})
+		if err != nil {
+			e2elog.Logf("error creating StorageClass %q in namespace %q: %v", sc.Name, sc.Namespace, err)
```
storageclass is not a namespaced object, remove the namespace from it?

nixpanic replied:
yeah, but it is part of the CephFS function too, this is just a copy/paste

Madhu-1 replied:
Maybe we can correct it both places in the same PR or in a different PR?

nixpanic replied:
this was done, btw

Madhu-1 commented on e2e/rbd_helper.go:
```diff
@@ -167,9 +168,25 @@ func createRBDStorageClass(
+			if isRetryableAPIError(err) {
+				return false, nil
+			}
+			if apierrs.IsNotFound(err) {
+				return false, nil
+			}
```
during create operation you will never get notFound error isn't it?

nixpanic replied:
probably not?

Madhu-1 replied:
Even I think so in that case we can remove this one.

nixpanic replied:
and this too

nixpanic committed:
e2e: cleanup StorageClass create retry

StorageClasses are cluster resources, not namespaced; there is no need
to log the namespace of a StorageClass.

When creating a StorageClass, NotFound is not an error that will be
returned, not need to check for it.

Signed-off-by: Niels de Vos <ndevos@redhat.com>

nixpanic commented:
/retest ci/centos/k8s-e2e-external-storage/1.22

nixpanic commented:
/retest ci/centos/mini-e2e-helm/k8s-1.23

nixpanic commented:
> /retest ci/centos/k8s-e2e-external-storage/1.22

Failed to create Ceph cluster

nixpanic commented:
> /retest ci/centos/mini-e2e-helm/k8s-1.23

node-plugin went missing:
```
Apr 14 16:48:14.577: INFO: 1 / 1 pods ready in namespace 'cephcsi-e2e-be428dbb301f' in daemonset 'csi-rbdplugin' (0 seconds elapsed)
Apr 14 16:48:14.588: FAIL: failed to run uname cmd : podlist is empty, stdErr:  
```

Madhu-1 commented:
/retest ci/centos/mini-e2e-helm/k8s-1.23

Madhu-1 commented:
> /retest ci/centos/mini-e2e-helm/k8s-1.23

Rook deployment failure

nixpanic commented:
@mergifyio rebase

nixpanic commented:
@mergifyio requeue
